daemon/api_prompting.go

```diff
@@ -97,6 +100,178 @@ func getUserID(r *http.Request) (uint32, Response) {
 	return uint32(userIDInt), nil
 }
 
+type invalidReason string
+
+const (
+	unsupportedValueReason invalidReason = "unsupported-value"
+	parseErrorReason       invalidReason = "parse-error"
+)
+
+type invalidFieldValue struct {
+	Reason   invalidReason `json:"invalid-reason"`
+	Metadata interface{}   `json:"metadata,omitempty"`
```

Comment:
we should flatten this and have an  "Value" and "Supported" fields here directly, both optional